gnd/src/main/java/com/google/android/gnd/repository/OfflineBaseMapRepository.java

```diff
@@ -256,4 +258,55 @@ public Completable deleteArea(String offlineAreaId) {
                     .andThen(localDataStore.deleteTileByUrl(tile)))
         .andThen(localDataStore.deleteOfflineArea(offlineAreaId));
   }
+
+  /**
+   * Retrieve all tile sources from a GeoJSON basemap specification, regardless of their
+   * coordinates.
+   */
+  public Single<ImmutableList<TileSource>> getAllActiveTileSources() {
+    // TODO: Only load tiles for basemap sources that are "active" (we will implement support for
+    // turing sources on and off).
+    return projectRepository
+        .getProjectLoadingState()
+        .compose(Loadable::values)
+        .map(Project::getOfflineBaseMapSources)
+        .doOnError(
+            throwable -> Timber.e(throwable, "No basemap sources specified for the active project"))
+        .map(ImmutableList::asList)
+        .flatMap(Flowable::fromIterable)
+        .firstOrError()
+        .map(this::identifyAndHandleSource)
+        .doOnError(
+            throwable ->
+                Timber.e(throwable, "Couldn't retrieve basemap sources for the active project"));
+  }
+
+  private ImmutableList<TileSource> identifyAndHandleSource(
+      OfflineBaseMapSource offlineBaseMapSource) throws IOException {
+    switch (offlineBaseMapSource.getType()) {
+      case GEOJSON:
+        File tileFile = downloadOfflineBaseMapSource(offlineBaseMapSource);
```

Comment:
re: downloading, that makes sense. Instead we should pass footprints source URLs to a remote provider.

As to the second point, that we don't need a footprints file at all, sure, but that means we're changing some of the assumptions we've made up to this point (at least how I've understood them).

When we get a footprint file, the notion is that the URLs are specifically provided to us and may or may not follow our parameterization strategy, and possibly could even point to other servers (e.g. in a distributed storage situation).

We can do what you're suggesting here, but I believe that would effectively mean that we assume the tile sources in a footprints file also follow our convention, at which point a footprints file is kind of redundant and purposeless anyway, right?

Reply:
Even if we standards the mbtiles paths, some kind of index file will be necessary if the client wants to know which ones exist. But since we're still discussing different options here let's punt on this for now.